server/channels/api4/post_test.go

```diff
@@ -38,70 +38,75 @@ func TestCreatePost(t *testing.T) {
 	defer th.TearDown()
 	client := th.Client
 
-	post := &model.Post{
-		ChannelId: th.BasicChannel.Id,
-		Message:   "#hashtag a" + model.NewId() + "a",
-		Props:     model.StringInterface{model.PropsAddChannelMember: "no good"},
-		DeleteAt:  101,
+	basicPost := func() *model.Post {
+		p := &model.Post{
+			ChannelId: th.BasicChannel.Id,
+			Message:   "#hashtag a" + model.NewId() + "a",
+			DeleteAt:  101,
+		}
+		p.AddProp(model.PropsAddChannelMember, "no good")
```

Comment:
Any reason to move this from a field to a method?

Reply:
The `Props` fields is marked as deprecated. While it makes no function difference to use the method, I think it's good practice to not use deprecated fields.